ci-operator/step-registry/idp/external-oidc/keycloak/aks/server/idp-external-oidc-keycloak-aks-server-commands.sh

```diff
@@ -0,0 +1,395 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+export KUBECONFIG="${SHARED_DIR}/kubeconfig"
+
+# Insall helm binary
+wget https://get.helm.sh/helm-v3.17.2-linux-amd64.tar.gz -O /tmp/helm.tar.gz
+tar -xvf /tmp/helm.tar.gz -C /tmp/
+chmod +x /tmp/linux-amd64/helm
+
+KEYCLOAK_ADMIN_TEST_USER="admin-$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 6 | head -n 1 || true)"
+KEYCLOAK_ADMIN_TEST_PASSWORD="$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 12 | head -n 1 || true)"
+KEYCLOAK_PREFIX="keycloak-$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 9 | head -n 1 || true)"
+KEYCLOAK_HOST="$KEYCLOAK_PREFIX.$HYPERSHIFT_BASE_DOMAIN"
+CLUSTER_CONSOLE="$(oc whoami --show-console)"
+CONSOLE_CLIENT_SECRET_VALUE=$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 32 | head -n 1 || true)
+
+# Generate random test users for testing
+TEST_USERS=""
+for i in $(seq 1 50);
+do
+    username="keycloak-testuser-${i}"
+    password=$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 12 | head -n 1 || true)
+    TEST_USERS+="${username}:${password},"
+done
+# The users will be mounted into keycloak pods and the real test users will be created there
+TEST_USERS=${TEST_USERS%?}
+
+# Install NGINX Ingress Controller
+/tmp/linux-amd64/helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+/tmp/linux-amd64/helm repo update
+/tmp/linux-amd64/helm install ingress-nginx ingress-nginx/ingress-nginx \
+  --namespace ingress-nginx \
+  --create-namespace \
+  --set controller.service.annotations."service\.beta\.kubernetes\.io/azure-load-balancer-health-probe-request-path"=/healthz \
+  --wait
+
+# Install cert-manager
+oc create namespace cert-manager || true
+oc apply -f https://github.com/jetstack/cert-manager/releases/download/v1.13.0/cert-manager.yaml
+oc wait --for=condition=ready pod -l app.kubernetes.io/instance=cert-manager -n cert-manager --timeout=300s
+# Create Let's Encrypt ClusterIssuer
+cat <<EOF | oc apply -f -
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-prod
+spec:
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
```

Comment:
Per my experience when I owned the cert-manager component before (now it transfers to OpenShift App Platform QE subteam), `acme-v02.api.letsencrypt.org/directory` is better used for end users' Product envs. It has rate limit to be called. https://letsencrypt.org/docs/rate-limits/ is about the rate limit. So I guess jobs may flaky fail due to the rate limit. And it says "We recommend running tests against our [staging API](https://letsencrypt.org/docs/staging-environment/) first". OAP QE subteam also use https://github.com/openshift/openshift-tests-private/blob/master/test/extended/oap/certmanager.go#L33 the staging.